plugins/inventory/flightctl.py

```
@@ -0,0 +1,301 @@
+    continue_token = None
+    while True:
+        try:
+            # Call list_devices passing the continuation token and other parameters
+            response = list_data(
+                var_continue=continue_token,  # Pass the current continuation token, if any
+                label_selector=label_selector,
+                field_selector=field_selector,
+                limit=limit
+            )
+        except Exception as e:
+            raise FlightctlApiException(f"Error retrieving data from FlightCTL API: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)